Add `LSP-prisma`.

Ultra-Instinct-05 opened the pull request:
This PR adds `LSP-prisma` (https://github.com/Sublime-Instincts/LSP-prisma)

`LSP-prisma` is a LSP helper package that uses the `Prisma Language Tools` to provide LSP support for Prisma schema files.

Jolg42 reviewed (approved):
Thanks @Ultra-Instinct-05 for the amazing work!

Note: we discussed this change in https://github.com/prisma/prisma/discussions/12316

predragnikolic commented:
Hello @Ultra-Instinct-05 

1. I would suggest to add a sentence in the README [installations steps](https://github.com/Sublime-Instincts/LSP-prisma#package-control),
that the user also has to install [Prisma](https://packagecontrol.io/packages/Prisma) syntax highlighting. 

2. I would suggest remove this paragraph from the README, 
as it might confuse users that they need to `Add Repository` even when the plugin is published to package control:

> Use Package Control: Add Repository from the command palette. Copy the github url (without the .git at the end) and enter it into the input panel that pops up at the bottom when you select Package Control: Add Repository. Now use Package Control: Install Package and search for LSP-prisma and install it.

Beside that everything looks good.

predragnikolic commented:
Also would you consider moving LSP-prisma to the SublimeLSP organization?

Ultra-Instinct-05 commented:
👋 @predragnikolic 

> I would suggest to add a sentence in the README [installations steps](https://github.com/Sublime-Instincts/LSP-prisma#package-control),
that the user also has to install [Prisma](https://packagecontrol.io/packages/Prisma) syntax highlighting.

Nice catch ! I have added that they need to install the highlighting package as well.

> I would suggest remove this paragraph from the README,
as it might confuse users that they need to Add Repository even when the plugin is published to package control:

I was going to do it *after* being added because otherwise it's going to be confusion that it's mentioned to use that but they can't find `LSP-prisma`. But oh well, might as well do it now. I have not removed the other way (`Add Repository`) because it's a valid way of installing but I have added that it has its own caveats and they should really just stick to installing using PC the normal way.

> Also would you consider moving LSP-prisma to the SublimeLSP organization?

Any chance I can continue with it in my own org ? It'll make things easier for me and a GitHub url change will probably break stuff for users that have already used `Add Repository`. I don't think there should be an issue with discoverability as it follows all conventions (hopefully) of a LSP helper package so searching it on PC should be easy.

predragnikolic commented:
> Any chance I can continue with it in my own org ? 

Yes, that is ok. 
I asked only because with SublimeLSP org there are more people who can jump in if needed, dependabot is already setup to create PR when a language server is updated.

----

You can also tag a new release, so others will see the updated readme on package control.

predragnikolic commented:
https://github.com/Sublime-Instincts/LSP-prisma/blob/da1cb77a8404ffd5b6e8be7d1a0bfe6cba04dfa0/LSP-prisma.sublime-settings#L8

langageID doesn't exist, instead there was languageId for ST 3.
Because you specified support ST 4, 
you could remove languageId, scopes and syntaxes in favor of just one setting, and that is `scopes`. (Here is an example https://github.com/sublimelsp/LSP-typescript/blob/12753b8ee6665e618b51b51b0b25970a966bfc8c/LSP-typescript.sublime-settings#L82 )

Ultra-Instinct-05 commented:
> Because you specified support ST 4,
you could remove languageId, scopes and syntaxes in favor of just one setting, and that is scopes

So I can remove the entire `languages` setting & only `selector` is enough ? I can verify it still continues to work with those removed, so I'll remove it.

I'll also push a new release for the changes.

predragnikolic commented:
> So I can remove the entire languages setting & only selector is enough ? 

`selector` was introduced in LSP for ST4, and it replaces the need for `languages`.

For me this looks good.

I'll merge this tomorrow,
just to give time to others to look at this as well.

rwols commented:
Shouldn't we wait until https://github.com/wbond/package_control_channel/pull/8498 is merged and available on PC?

Ultra-Instinct-05 commented:
It's technically not an issue. The reason is because the `Prisma` package available on PC provides the selector `source.prisma` which is all that is needed for `LSP-prisma` to work, right ? Of course, the `Prisma` package is uses the `.tmLanguage` format and the one in the PR uses the newer format (with other good stuff). So as long as the users have that `Prisma` package installed, they can use `LSP-prisma` (Of course with not so good highlighting)

predragnikolic commented:
Thanks!
